be/src/storage/rowset/segment_v2/bitshuffle_page.h

```diff
@@ -387,16 +389,31 @@ class BitShufflePageDecoder final : public PageDecoder {
 
 template <FieldType Type>
 inline Status BitShufflePageDecoder<Type>::next_batch(size_t* count, vectorized::Column* dst) {
+    vectorized::SparseRange read_range;
+    size_t begin = current_index();
+    read_range.add(vectorized::Range(begin, begin + *count));
+    RETURN_IF_ERROR(next_batch(read_range, dst));
+    *count = current_index() - begin;
+    return Status::OK();
+}
+
+template <FieldType Type>
+inline Status BitShufflePageDecoder<Type>::next_batch(const vectorized::SparseRange& range, vectorized::Column* dst) {
     DCHECK(_parsed);
     if (PREDICT_FALSE(_cur_index >= _num_elements)) {
-        *count = 0;
         return Status::OK();
     }
 
-    *count = std::min(*count, static_cast<size_t>(_num_elements - _cur_index));
-    int n = dst->append_numbers(get_data(_cur_index * SIZE_OF_TYPE), *count * SIZE_OF_TYPE);
-    DCHECK_EQ(*count, n);
-    _cur_index += *count;
+    size_t nread = std::min(static_cast<size_t>(range.span_size()), static_cast<size_t>(_num_elements - _cur_index));
```

Comment:
better rename nread to to_read or just count.
nread means how many element read in my opinion.